### PR TITLE
Limit orders fees

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

#Fixes: https://github.com/cowprotocol/cowswap/issues/1348

This PR shows the actual fee also for Limit Orders.

The main changes are to add the new field `executedSurplusFee` from the API and apply it for the different calculations. 

Before:
image

After:
image

## Not included
Note that there's a bit of repetition, for how the order info is derived for getting the different sections of the explorer, I didn't aim to do a refactor of that in this PR

I didn't do also storybook/tests in this PR

# To Test

1.  Load a Limit Order in DEV: https://protocol-explorer.dev.gnosisdev.com/goerli/orders/0x1473e457f1519494e72cdba6fda965dfdd7ea07320dad844a5461b608158c35b79063d9173c09887d536924e2f6eadbabac099f56388a9c1
2. Open the same one in this PR: look for `0x1473e457f1519494e72cdba6fda965dfdd7ea07320dad844a5461b608158c35b79063d9173c09887d536924e2f6eadbabac099f56388a9c1` 
3. Verify all fields
4. Make sure SWAPS are not broken :) 

# Background

Limit Orders have fee 0, the fee is payed from the surplus. The explorer is transparent showing the value of this fee.

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @anxolin , I can see than an order is Partially filled: 
image
And there is no Fees field

File: src/utils/operator.ts [hunk 1]
```diff
@@ -12,14 +12,17 @@ import { formattingAmountPrecision, formatSmartMaxPrecision } from 'utils'
...
 
 function isOrderFilled(order: RawOrder): boolean {
+  const { kind, executedBuyAmount, sellAmount, executedSellAmount, buyAmount, executedFeeAmount, executedSurplusFee } =
+    order
   let amount, executedAmount
 
-  if (order.kind === 'buy') {
-    amount = new BigNumber(order.buyAmount)
-    executedAmount = new BigNumber(order.executedBuyAmount)
+  const totalFee = new BigNumber(executedFeeAmount).plus(new BigNumber(executedSurplusFee))
```
Comment (alfetopito) [inline src/utils/operator.ts:19]: Not for this PR, but what about returning the `total` fee from the SDK directly and hide this logic?

**Review by alfetopito — APPROVED** [review]

File: src/utils/operator.ts (hunk 1 above)
Comment (anxolin) [reply src/utils/operator.ts:19]: yes, we have a lot of duplication 😓

**Comment by elena-zh** [comment]
Hey @anxolin , great job!
But now Execution price and Fees are not displayed to a market order:
image



One more question: now I see that Execution price is different in CoW Swap and on Explorer. Should it be fixed here or in CoW Swap?
execution price

Thanks!

File: src/utils/operator.ts [hunk 2]
```diff
@@ -173,14 +176,16 @@ export function getOrderSurplus(order: RawOrder): Surplus {
...
  */
 export function getOrderExecutedAmounts(
-  order: Pick<RawOrder, 'executedBuyAmount' | 'executedSellAmount' | 'executedFeeAmount'>,
+  order: Pick<RawOrder, 'executedBuyAmount' | 'executedSellAmount' | 'executedFeeAmount' | 'executedSurplusFee'>,
 ): {
   executedBuyAmount: BigNumber
   executedSellAmount: BigNumber
 } {
+  const { executedBuyAmount, executedSellAmount, executedFeeAmount, executedSurplusFee } = order
+  const totalFee = new BigNumber(executedFeeAmount).plus(executedSurplusFee)
```
Comment (shoom3301) [inline src/utils/operator.ts:185]: Considering https://github.com/cowprotocol/cowswap/pull/1588#discussion_r1037501631 it should be:

```typescript
const totalFee = new BigNumber(executedSurplusFee ?? executedFeeAmount)
```

File: src/utils/operator.ts (hunk 1 above)
Comment (shoom3301) [reply src/utils/operator.ts]: @anxolin If you wish I can help with it

Comment (shoom3301) [reply src/utils/operator.ts]: https://github.com/cowprotocol/cow-sdk/pull/88 - anyway, I took 2 mins to add to the SDK :)

File: src/utils/operator.ts (hunk 2 above)
Comment (anxolin) [reply src/utils/operator.ts:185]: There's an open discussion with this, yes,I will update with the resolution of it

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — CHANGES_REQUESTED** [review]
Filled limit orders are in Open status on Explorer (when expiry date is in the future): 
image
And partially filled when expiry date is in the past:
image

Expected: orders should be filled.

Execution price don't match, and there is no Fee field for filled limit orders
image


Account details for tests: https://pr265--explorer.review.gnosisdev.com/gc/address/0x9fa3c00a92ec5f96b1ad2527ab41b3932efeda58

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by elena-zh — APPROVED** [review]
LGTM!

**Comment by anxolin** [comment]
Thanks a lot @shoom3301 ❤️

**Pull request merged**